stc_game/Assets/Scripts/CharacterBehavior/DecisionScripts/CheckDeathDecision.cs

```diff
@@ -30,6 +31,10 @@ private void Die(StateController controller)
         if (controller.gameObject.CompareTag("Enemy"))
         {
             controller.target.GetComponent<CharacterStats>().stats.GainXP(controller.characterStats.stats.XP);
+            if (controller.gameObject.GetComponent<SetQuestProgressOnKilled>())

```

Comment:
How does this work? I'm guessing mob that IS a quest target gets a SetProgessOnKilled object attached to him from the scene in the unity editor, then when he dies, if that object exists on him then it gets triggered?

Reply:
Yes that’s correct.. if you want we can wait on this pull request until the next meeting so we can talk further about this bug fix !